chips/sam4l/src/acifc.rs

```diff
@@ -0,0 +1,370 @@
+			InterruptAfterEvaluation = 3,
+			// Window interrupt when the common input voltage enters the window (i.e., rising-edge of ACWOUT)
+			InterruptEnterWindow = 4,
+			// Window interrupt when the common input voltage leaves the window (i.e., falling-edge of ACWOUT)
+			InterruptLeaveWindow = 5
+	]
+	],
+
+	ACConfiguration [
+		// If equal to one, AC is always enabled. 
+		ALWAYSON OFFSET(27) NUMBITS(1) [],
+		// 0: Low-power mode. 1: Fastm ode.
+		FAST OFFSET(26) NUMBITS(1) [],
+		// Hysteresis voltage value: 0/25/50/75 mV
+		HYS OFFSET(24) NUMBITS(2) [
+			HysteresisVoltage0mV = 0,
+			HysteresisVoltage25mV = 1,
+			HysteresisVoltage50mV = 2,
+			HysteresisVoltage75mV = 3
+		],
+		// Setting this to one will output peripheral event when ACOUT is zero.
+		EVENP OFFSET(17) NUMBITS(1) [],
+		// Setting this to one will output peripheral event when ACOUT is one.
+		EVENN OFFSET(16) NUMBITS(1) [],
+		// Negative input select. 00: ACANx pint selected, others reserved.
+		INSELN OFFSET(8) NUMBITS(2) [],
+		// Choose between analog comparator mode.
+		MODE OFFSET(4) NUMBITS(2) [
+			Off = 0,
+			ContinuousMeasurementMode = 1,
+			// User Triggered Single Measurement Mode
+			UserMode = 2,
+			// Peripheral Event Single Measurement Mode
+			PeripheralMode = 3
+		],
+		// Interrupt settings
+		IS OFFSET(0) NUMBITS(2) [
+			// When Vinp > Vinn
+			WhenVinpGtVinn = 0,
+			// When Vinp < Vinn
+			WhenVinpLtVinn = 1,
+			// On toggle of ACOUT
+			OnToggleOfACOUT = 2,
+			// When comparison of Vinp and Vinn is done
+			WhenComparisonDone = 3
+		]
+	]
+];
+
+const BASE_ADDRESS: *mut AcifcRegisters = 0x40040000 as *mut AcifcRegisters;
+
+pub struct Acifc {
+    registers: *mut AcifcRegisters,
+    //client: Option<&'a acifc::Client>,
+}
+
+pub static mut ACIFC: Acifc = Acifc::new();
+
+// Implement constructor for struct Acifc
+impl Acifc {
+    const fn new() -> Self {
+        Acifc {
+            registers: BASE_ADDRESS,
+            //client: Cell::new(None),
+        }
+    }
+
+    fn enable_clock(&self) {
+        unsafe {
+            pm::enable_clock(pm::Clock::PBA(pm::PBAClock::ACIFC));
+        }
+    }
+
+    fn disable_clock(&self) {
+        unsafe {
+            pm::disable_clock(pm::Clock::PBA(pm::PBAClock::ACIFC));
+        }
+    }
+
+    pub fn set_client(&self) -> ReturnCode {
+        ReturnCode::SUCCESS
+    }
+
+    // Functions which (should) enable interrupts for the window or startup modes
+    fn enable_interrupts(&self) {
+        let regs: &AcifcRegisters = unsafe { &*self.registers };
+        regs.ier
+            .write(Interrupt::ACINT0::SET + Interrupt::ACINT1::SET);
+    }
+
+    fn disable_interrupts(&self) {
+        let regs: &AcifcRegisters = unsafe { &*self.registers };
+        regs.idr
+            .write(Interrupt::ACINT0::SET + Interrupt::ACINT1::SET);
+    }
+
+    // Handling interrupts not yet implemented.
+    pub fn handle_interrupt(&mut self) {}
```

Comment:
You can use rust's `unimplemented!("ACIFC Interrupt");` in the function body here (will print nice error if it ever gets called somehow)